lib/dry/system/components/external.rb

```diff
@@ -3,10 +3,68 @@
 module Dry
   module System
     module Components
+      class Config
+        def self.new(&block)
+          config = super
+          yield(config)
+          config
+        end
+
+        def initialize
+          @settings = {}
+        end
+
+        def to_hash
+          @settings
+        end
+
+        private
+
+        def method_missing(meth, value = nil)
+          if meth.to_s.end_with?('=')
+            @settings[meth.to_s.gsub('=', '').to_sym] = value
+          elsif @settings.key?(meth)
+            @settings[meth]
+          else
+            super
+          end
+        end
+      end
+
       class External < Components::Bootable
+        attr_reader :triggers
+
+        attr_reader :config
+
+        def initialize(*)
+          super
+          @triggers = { before: Hash.new { |h, k| h[k] = [] }, after: Hash.new { |h, k| h[k] = [] } }
+          instance_exec(&finalize)
+        end
+
+        def trigger(key, event, lf_container)
+          triggers[key][event].each do |fn|
+            container.instance_exec(lf_container, &fn)
+          end
+
+          self
+        end
+
+        def after(event, &block)
+          triggers[:after][event] << block
+          self
+        end
+
         def external?
           true
         end
+
+        def configure(&block)
+          @config = Config.new do |config|
```

Comment:
Might be missing something but I think this could just be `Config.new(&block)`?

Reply:
@AMHOL fixed, thanks (: